tests/validation/domain_test.go

```diff
@@ -14,33 +15,85 @@ func domainTests(t *testing.T, when spec.G, it spec.S) {
 
 	var service testutil.TestService
 	var domain testutil.TestDomain
+	var response *route53.ChangeResourceRecordSetsOutput
+
+	it.Before(func() {
+		service.Create(t, "ibuildthecloud/demo:v1")
+	})
 
 	it.After(func() {
 		service.Remove()
 	})
 
-	when("get a cluster domain", func() {
+	when("a cname is registered with a service", func() {
+		var domainName string
+
 		it.Before(func() {
-			service.Create(t, "strongmonkey1992/autoscale:testing")
+			domainName, _ = service.GetKubeFirstClusterDomain()
+			response = testutil.CreateRoute53DNS("CNAME", domainName)
 		})
 
 		it.After(func() {
 			domain.UnRegister()
 			if os.Getenv("RIO_CNAME") != "" {
-				testutil.DeleteCNAME(service.GetKubeFirstClusterDomain())
+				testutil.DeleteRoute53DNS(domainName)
 			}
 		})
 
-		it("should create a new DNS CNAME for the cluster domain and retrieve the service content using it", func() {
-			response := testutil.CreateCNAME(service.GetKubeFirstClusterDomain())
-			assert.Equal(t, *response.ChangeInfo.Comment, "Add CNAME to Rio Cluster Domain")
+		it("should allow retrieval of the service content using it", func() {
+			assert.Equal(t, "DNS for custom record in Rio Cluster Domains", *response.ChangeInfo.Comment)
 			domain.RegisterDomain(t, testutil.GetCNAMEInfo(), service.Name)
 			assert.Equal(t, testutil.GetCNAMEInfo(), domain.GetDomain())
 			assert.Equal(t, testutil.GetCNAMEInfo(), domain.GetKubeDomain())
 			urlHTTPResponse, _ := testutil.WaitForURLResponse("http://" + testutil.GetCNAMEInfo())
-			assert.Equal(t, "Hi there, I am rio:production6", urlHTTPResponse)
+			assert.Equal(t, "Hello World", urlHTTPResponse)
 			urlHTTPSResponse, _ := testutil.WaitForURLResponse("https://" + testutil.GetCNAMEInfo())
-			assert.Equal(t, "Hi there, I am rio:production6", urlHTTPSResponse)
+			assert.Equal(t, "Hello World", urlHTTPSResponse)
+		})
+	}, spec.Parallel())
+
+	when("a clusterdomain is added", func() {
+		var ip string
+		var otherService testutil.TestService
+
+		it.Before(func() {
+			_, ip = service.GetKubeFirstClusterDomain()
+			response = testutil.CreateRoute53DNS("A", ip)
+			testutil.CreateSelfSignedCert()
+			_, err := testutil.KubectlCmd([]string{"-n", "rio-system", "create", "secret", "tls",
+				testutil.GetAInfo() + "-tls", "--cert=testcert.pem", "--key=testkey.pem"})
+			if err != nil {
+				assert.Fail(t, "Failed to create tls secret for custom cluster domain: "+err.Error())
+			}
+			domain.ApplyClusterDomain(ip)
+			otherService.Create(t, "ibuildthecloud/demo:v3")
+		})
+
+		it.After(func() {
+			otherService.Remove()
+			if os.Getenv("RIO_A_RECORD") != "" {
+				testutil.DeleteRoute53DNS(ip)
+			}
```

Comment:
Do you need to clean up cert ?

Reply:
Ah yes I probably should. Added them to gitignore but will fix ipOrDns fmt issue and clean these up